### PR TITLE
Closes #8962: Add Tab Counter tests

### LINKED ISSUE #8962: [XCUITest] Add a check to verify the number of tabs open in the Tab Tray
Opening this task to add a verification of the # of tabs open in tab tray, when it is open, after removing a tab and different scenarios that would be useful to have cover and detect issues like #8955

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-3043)

### PR DESCRIPTION
**Pull request by AaronMT**
For #8962

### CODE REVIEW

**Comment by AaronMT** [comment]
https://github.com/mozilla-mobile/firefox-ios/pull/9142/checks?check_run_id=3653986670
> BookmarkingTests.testBookmarkLibraryAddDeleteBookmark()

File: XCUITests/TabCounterTests.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        navigator.createNewTab()
+        navigator.nowAt(NewTabScreen)
+        navigator.performAction(Action.CloseURLBarOpen)
+        waitForTabsButton()
+
+        tabsOpen = app.buttons["Show Tabs"].value
+        XCTAssertEqual("2", tabsOpen as? String)
+
+        navigator.goto(TabTray)
+        navigator.performAction(Action.AcceptRemovingAllTabs)
```
Comment (isabelrios) [inline XCUITests/TabCounterTests.swift:43]: can you please add a step before this one in which you only remove one tab (using the 'x' button) in the tab tray and check that the counter decrements? that's the issue we have seen several times, that removing tab by tab the counter is not correctly updated.
Thanks!

Comment (AaronMT) [reply XCUITests/TabCounterTests.swift]: Replaced the close all tabs with a close button action instead.

File: XCUITests/TabCounterTests.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        tabsOpen = app.buttons["Show Tabs"].value
+        XCTAssertEqual("2", tabsOpen as? String)
+    }
+
+    func testTabDecrement() throws {
+        navigator.nowAt(NewTabScreen)
+        navigator.performAction(Action.CloseURLBarOpen)
+        waitForTabsButton()
+
+        var tabsOpen = app.buttons["Show Tabs"].value
```
Comment (isabelrios) [inline XCUITests/TabCounterTests.swift:31]: I should have realized about that before and explain the purpose of the test case better...apologies... the problem where we have seen regressions is with the counter in the tab tray, not the counter in the homescreen.
This check is verifying the counter in the homescreen but the one we would like to check is this one in the tab tray
Sin título
:

And same for all the other checks

**Review by isabelrios — CHANGES_REQUESTED** [review]
Sorry @AaronMT I should have noticed before where the tab counter check is being verified. We already have tests checking the tab counter in Homescreen, but not in the Tab Tray. That's the part we would like to cover with these tests, check the tab counter in the tab tray after adding/removing tabs. 
I should have specified that clearer, sorry about that.

File: XCUITests/TabCounterTests.swift (hunk 2 above)
Comment (AaronMT) [reply XCUITests/TabCounterTests.swift:36]: Added this along the original check. The button's only label is the counter number itself so it's a little messy in the way it's accessed if that's ok.

**Comment by isabelrios** [comment]
@AaronMT you may need to pull latest changes on main to get this green as the workflow was broken due to bitrise changes

**Pull request merged**